src/olympia/files/models.py

```diff
@@ -68,9 +68,9 @@ def files_upload_to_callback(instance, filename):
     a filename, but the filename is completely ignored here (it's meant to
     represent the user-provided filename in user uploads).
     """
-    # slugify drops unicode so we may end up with an empty string.
-    # Apache did not like serving unicode filenames (bug 626587).
-    name = slugify(instance.addon.name).replace('-', '_') or 'addon'
+    # Start with the add-on slug, but make it go through django's slugify() to
+    # drop unicode characters.
```

Comment:
_is_ there still a problem with filename's with unicode characters in them?  The original comment referenced Apache and the bug talks about release mirrors, both of which aren't a thing in our current stack, iirc.

Reply:
It's probably fine, and we do have a few tests that prove we know how to deal with unicode filenames if necessary, but then again, non-ascii in things like `Content-Disposition` header is a mess, so I don't want to change that now...